projects/packages/my-jetpack/_inc/components/welcome-flow/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { Container, Col, Button } from '@automattic/jetpack-components';
+import { __ } from '@wordpress/i18n';
+import { close } from '@wordpress/icons';
+import React, { useCallback, useEffect } from 'react';
+import useWelcomeBanner from '../../data/welcome-banner/use-welcome-banner';
+import useAnalytics from '../../hooks/use-analytics';
+import useMyJetpackConnection from '../../hooks/use-my-jetpack-connection';
+import { CardWrapper } from '../card';
+import ConnectionStep from './ConnectionStep';
+import styles from './style.module.scss';
+
+const WelcomeFlow: React.FC = () => {
+	const { recordEvent } = useAnalytics();
+	const { isWelcomeBannerVisible, dismissWelcomeBanner } = useWelcomeBanner();
+	const { siteIsRegistered, siteIsRegistering, handleRegisterSite } = useMyJetpackConnection( {
+		skipUserConnection: true,
+	} );
+	const [ visible, setVisible ] = React.useState( isWelcomeBannerVisible );
+
+	const onDismissClick = useCallback( () => {
+		recordEvent( 'jetpack_myjetpack_welcome_banner_dismiss_click' );
+		setVisible( false );
+		dismissWelcomeBanner();
+	}, [ recordEvent, dismissWelcomeBanner ] );
+
+	useEffect( () => {
+		// TODO: It's a temporary code instead of Survey step
+		if ( siteIsRegistered && visible ) {
+			setVisible( false );
+			dismissWelcomeBanner();
+		}
+	}, [ visible, siteIsRegistered, dismissWelcomeBanner ] );
+
+	if ( ! visible ) {
+		return null;
+	}
+
+	return (
+		<Container horizontalSpacing={ 3 } className={ styles[ 'banner-container' ] }>
+			<Col lg={ 12 } className={ styles.banner }>
+				<CardWrapper className={ styles[ 'banner-card' ] }>
+					<Container
+						horizontalSpacing={ 0 }
+						horizontalGap={ 0 }
+						className={ styles[ 'banner-content' ] }
+					>
+						{ ! siteIsRegistered ? (
```

Comment:
In the previous work, we were also displaying the Welcome banner if the site was site-connected but they had never seen the welcome banner before. Here, we only display it if the site is not connected. Is that change expected?

Reply:
> Here, we only display it if the site is not connected. Is that change expected?

It's because of the hook I've added:
```
	useEffect( () => {
		// TODO: It's a temporary code instead of Survey step
		if ( siteIsRegistered && visible ) {
			setVisible( false );
			dismissWelcomeBanner();
		}
	}, [ visible, siteIsRegistered, dismissWelcomeBanner ] );
```

It will not be part of the final solution, and I don't think this PR should be merged without the following ones - but rather I treat it as a base for changes.

The conceptual idea here is:
1. Site disconnected and banner not dismissed: Welcome Flow -> Connection step
2. Site connected and banner not dismissed: Welcome Flow -> Survey step
3. Banner dismissed: no Welcome Flow